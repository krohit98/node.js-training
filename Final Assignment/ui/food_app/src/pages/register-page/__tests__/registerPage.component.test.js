import { render, screen } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import RegisterPage from '../registerPage.component';
import RegisterForm from '../../../components/register-form/registerForm.component';
import * as reactRedux from 'react-redux';

jest.mock('../../../components/register-form/registerForm.component');

describe("Unit testing RegisterPage component",()=>{

    test("Snapshot testing",()=>{

        jest.spyOn(reactRedux, "useDispatch").mockReturnValue(jest.fn());

        const renderer = new ShallowRenderer();
        renderer.render(<RegisterPage />);
        const result = renderer.getRenderOutput();

        expect(result).toMatchSnapshot();
    })

    test("Should render RegisterPage correctly",()=>{
        RegisterForm.mockImplementation(()=><div>RegisterForm</div>);
        render(<RegisterPage />);
        const heading = screen.getByText("Sign-Up for a New Account");
        const registerForm = screen.getByText("RegisterForm");
        expect(heading).toBeInTheDocument();
        expect(registerForm).toBeInTheDocument();
    })
})