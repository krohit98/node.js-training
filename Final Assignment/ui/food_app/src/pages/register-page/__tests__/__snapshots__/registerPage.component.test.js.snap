// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing RegisterPage component Snapshot testing 1`] = `
<div
  className="pageDiv"
>
  <h1
    className="display-3 pageHeading"
  >
    Sign-Up for a New Account
  </h1>
  <RegisterForm
    submitHandler={[Function]}
  />
  
</div>
`;
